pkg/controller/logstorage/initializer/initializing_controller.go

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) 2020-2023 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package initializer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/stringsutil"
+
+	kbv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/kibana/v1"
+	operatorv1 "github.com/tigera/operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/tigera/operator/pkg/controller/options"
+	"github.com/tigera/operator/pkg/controller/status"
+	"github.com/tigera/operator/pkg/controller/utils"
+	"github.com/tigera/operator/pkg/render"
+)
+
+var log = logf.Log.WithName("controller_logstorage")
+
+const (
+	DefaultElasticsearchStorageClass = "tigera-elasticsearch"
+	LogStorageFinalizer              = "tigera.io/eck-cleanup"
+	ResourceName                     = "log-storage"
+	defaultEckOperatorMemorySetting  = "512Mi"
+)
+
+// Add creates a new LogStorage Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, opts options.AddOptions) error {
+	if !opts.EnterpriseCRDExists {
+		return nil
+	}
+
+	// Create the reconciler
+	r := &LogStorageInitializer{
+		client:      mgr.GetClient(),
+		scheme:      mgr.GetScheme(),
+		multiTenant: opts.MultiTenant,
+		status:      status.New(mgr.GetClient(), ResourceName, opts.KubernetesVersion),
+	}
+	r.status.Run(opts.ShutdownContext)
+
+	// Create a controller using the reconciler and register it with the manager to receive reconcile calls.
+	c, err := controller.New("log-storage-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Configure watches for operator.tigera.io APIs this controller cares about.
+	if err = c.Watch(&source.Kind{Type: &operatorv1.LogStorage{}}, &handler.EnqueueRequestForObject{}); err != nil {
```

Comment:
I think this controller is also dependent on Installation{} and should watch it.